llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -437,6 +469,15 @@ Value *VPInstruction::generate(VPTransformState &State) {
     Value *Cond = State.get(getOperand(0));
     Value *Op1 = State.get(getOperand(1));
     Value *Op2 = State.get(getOperand(2));
+    auto *CondVec = dyn_cast<VectorType>(Cond->getType());
```

Comment:
This feels fragile. Perhaps better to change the place where the select is introduced (`LoopVectorizationPlanner::adjustRecipesForReductions`) if it's a partial reduction? I guess that may mean we need to store details of partial reductions in `LoopVectorizationPlanner` and pass those to `VPRecipeBuilder` in the constructor.

Reply:
It turns my understanding of how selects work with `<Y x i1>` conditions was entirely wrong so I'll revert these changes and leave supporting predicated loops until later.